grpc/server.go

```diff
@@ -43,8 +44,27 @@ func (s *Server) StreamSearch(req *v1.SearchRequest, ss v1.WebserverService_Stre
 		return status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	onMatch := stream.SenderFunc(func(res *zoekt.SearchResult) {
-		ss.Send(res.ToProto())
+	onMatch := stream.SenderFunc(func(r *zoekt.SearchResult) {
+		result := r.ToProto()
+
+		f := func(filesChunk []*v1.FileMatch) error {
+			return ss.Send(&v1.SearchResponse{
+				Stats:         result.GetStats(),
```

Comment:
Note: Need to figure out how stats / and progress are collected on the client - do they need to be adjusted when splitting them up?

Reply:
For example look at how stats and progress is used in https://github.com/sourcegraph/zoekt/pull/637 and in the sendByRepository function.

Progress and stats are tricky. Imagine you have chunks 1, 2, ... N.
```go
# Progress < N
Progress: {
  Priority: r.Progress.Priority,
  MaxPendingPriority: max(r.Progress.Priority, r.Progress.MaxPendingPriority),
},

# Progress N
Progress: r.Progress,
```

For stats I am much more unsure on the correct thing to do. I think you just send Stats for 1, then empty for everything else.